eng/native/tryrun.cmake

```diff
@@ -27,6 +27,10 @@ elseif(EXISTS /System/Library/CoreServices)
   set(DARWIN 1)
 elseif(EXISTS ${CROSS_ROOTFS}/etc/tizen-release)
   set(TIZEN 1)
+elseif(EXISTS ${CROSS_ROOTFS}/boot/system/develop/headers/config/HaikuConfig.h)
+  set(HAIKU 1)
+  set(CMAKE_SYSTEM_NAME Haiku)
```

Comment:
We do not set `CMAKE_SYSTEM_NAME` for other OSes (except FreeBSD). Why do we need to set it?